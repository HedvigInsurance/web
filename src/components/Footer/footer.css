.Footer {
  background-color: #0f007a;
  padding: 30px 20px 20px;
  display: block;
  flex-direction: row;
}

.Footer__start {
  flex: 1;
  padding-right: 20px;
}

.Footer__end {
  align-self: center;
  margin-top: 20px;
}

.Footer__links {
  justify-content: start;
  align-items: center;
}

.Footer__logo {
  padding-right: 30px;
}

.Footer__link {
  color: white;
  display: block;
}

.Footer__icon {
  height: 28px;
  width: 84px;
  background-image: url("/assets/identity/hedvig_wordmark/hedvig_wordmark_white.svg");
  background-size: 84px 28px;
  background-repeat: no-repeat;
}

.Footer__appLink:first-child {
  margin-right: 10px;
}

@media screen and (min-width: 700px) {
  .Footer {
    display: flex;
  }

  .Footer__end {
    display: flex;
    margin-top: 0;
  }
}
