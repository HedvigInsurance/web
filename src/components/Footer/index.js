import React from 'react';
import Link from 'gatsby-link';

import './footer.css';

const Footer = () => (
  <footer>
    <div className="Footer">
      <div className="Footer__start">
        <div className="Footer__links">
          <div className="Footer__logo">
            <a href="/">
              <div role="img" className="Footer__icon" aria-label="Hedvig logo" />
            </a>
          </div>
          <nav className="">
            <Link className="Footer__link" to="/contact">
              Kontakt
            </Link>
            <Link className="Footer__link" to="/legal">
              Legal&nbsp;information
            </Link>
          </nav>
        </div>
      </div>
    </div>
  </footer>
);

export default Footer;
