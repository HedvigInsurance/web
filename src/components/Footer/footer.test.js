import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from './index';

describe('<Footer />', () => {
  it('Should render correctly', () => {
    const component = shallow(<Footer />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
