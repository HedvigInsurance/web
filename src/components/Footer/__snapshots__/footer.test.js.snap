// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should hide app store badges when on /download page 1`] = `
<div
  className="u-backgroundPrimaryDarkBlue u-flexNone"
>
  <div
    className="Container"
  >
    <div
      className="u-md-flex u-lg-flex u-flexRow u-spaceMT5 u-spaceMB8"
    >
      <div
        className="u-flexGrow1"
      >
        <nav
          className="Grid"
        >
          <div
            className="u-md-size1of2 u-lg-size1of2 u-maxWidth1of3"
          >
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/faq"
            >
              Vanliga frågor
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/about-us"
            >
              Om Hedvig
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/contact"
            >
              Kontakt
            </GatsbyLink>
          </div>
          <div
            className="u-md-size1of2 u-lg-size1of2 u-maxWidth1of3"
          >
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/press"
            >
              Press
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/privacy"
            >
              Personuppgifter
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/legal"
            >
              Legal information
            </GatsbyLink>
          </div>
        </nav>
      </div>
    </div>
    <div
      className="u-spaceMB7"
    >
      <a
        className="u-spacePR10"
        href="https://www.facebook.com/hedvigers"
      >
        <ReactComponent />
      </a>
      <a
        className="u-spacePR9"
        href="https://www.instagram.com/hedvigers/"
      >
        <ReactComponent />
      </a>
      <a
        className="u-spacePR10"
        href="https://twitter.com/hedvigers"
      >
        <ReactComponent />
      </a>
    </div>
    <div
      className="u-spaceMB7"
    >
      <p
        className="u-fontSize10 u-colorWhite u-spaceMB9"
      >
        © Hedvig AB. Huvudkontor: Artillerigatan 10, 114 51, Stockholm. Org. nr. 559093-0334.
      </p>
      <p
        className="u-fontSize10 u-colorWhite u-spaceMB9"
      >
        Exklusiv försäkringsgivare för Hedvigs försäkringar är International Insurance Company of Hannover SE, Sverige filial, org. nr. 516402-6345. Hedvig står under Finansinspektionens tillsyn.
      </p>
    </div>
  </div>
</div>
`;

exports[`<Footer /> Should render correctly with app store downloads 1`] = `
<div
  className="u-backgroundPrimaryDarkBlue u-flexNone"
>
  <div
    className="Container"
  >
    <div
      className="u-md-flex u-lg-flex u-flexRow u-spaceMT5 u-spaceMB8"
    >
      <div
        className="u-flexGrow1"
      >
        <nav
          className="Grid"
        >
          <div
            className="u-md-size1of2 u-lg-size1of2 u-maxWidth1of3"
          >
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/faq"
            >
              Vanliga frågor
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/about-us"
            >
              Om Hedvig
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/contact"
            >
              Kontakt
            </GatsbyLink>
          </div>
          <div
            className="u-md-size1of2 u-lg-size1of2 u-maxWidth1of3"
          >
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/press"
            >
              Press
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/privacy"
            >
              Personuppgifter
            </GatsbyLink>
            <GatsbyLink
              className="u-spaceMB9 u-linkBlock u-colorWhite"
              to="/legal"
            >
              Legal information
            </GatsbyLink>
          </div>
        </nav>
      </div>
      <div
        className="u-md-textRight u-lg-textRight"
      >
        <a
          className="u-block u-spaceMB11"
          href="https://itunes.apple.com/se/app/hedvig/id1303668531?mt=8"
          id="cta-footer-download-apple-store"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Ladda ner på App Store"
            height={54}
            src="/assets/appstores/app-store-badge@2x.png"
          />
        </a>
        <a
          className="u-block u-spaceMB11"
          href="https://play.google.com/store/apps/details?id=com.hedvig.app"
          id="cta-footer-download-play-store"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Ladda ner på Google Play"
            height={54}
            src="/assets/appstores/google-play-badge@2x.png"
          />
        </a>
      </div>
    </div>
    <div
      className="u-spaceMB7"
    >
      <a
        className="u-spacePR10"
        href="https://www.facebook.com/hedvigers"
      >
        <ReactComponent />
      </a>
      <a
        className="u-spacePR9"
        href="https://www.instagram.com/hedvigers/"
      >
        <ReactComponent />
      </a>
      <a
        className="u-spacePR10"
        href="https://twitter.com/hedvigers"
      >
        <ReactComponent />
      </a>
    </div>
    <div
      className="u-spaceMB7"
    >
      <p
        className="u-fontSize10 u-colorWhite u-spaceMB9"
      >
        © Hedvig AB. Huvudkontor: Artillerigatan 10, 114 51, Stockholm. Org. nr. 559093-0334.
      </p>
      <p
        className="u-fontSize10 u-colorWhite u-spaceMB9"
      >
        Exklusiv försäkringsgivare för Hedvigs försäkringar är International Insurance Company of Hannover SE, Sverige filial, org. nr. 516402-6345. Hedvig står under Finansinspektionens tillsyn.
      </p>
    </div>
  </div>
</div>
`;
